// zj_time-lite/index.js
var moment = require('moment');
var dateFormat = 'YYYY-MM-DD HH:mm:ss';

exports.IdParseTime = (id, formatStr) => {
    return ' ' + moment(parseInt(
        id.substr(0, 8),
        16
    ) * 1000).format(formatStr ? formatStr : dateFormat)
};

exports.GetMonthBeginTime = (timeStr, formatStr) => {
    var year = moment(timeStr, formatStr ? formatStr : dateFormat).year();
    var month = moment(timeStr, formatStr ? formatStr : dateFormat).month() + 1;
    return moment(year + '-' + month + '-01', 'YYYY-MM-DD').unix();
};

exports.CurrentTimeStr = (time, format) => {
    if (time <= 0)
        return ' ';

    if (format)
        dateFormat = format;
    return ' ' + moment(new Date(time * 1000)).format(dateFormat);
};

exports.CurrentTimeStamp = () => {
    var stamp = Math.floor(Date.now() / 1000);
    return stamp;
};

exports.ConvertTimeStamp = (timeStr, formatStr) => {
    var stamp = moment(timeStr, formatStr ? formatStr : dateFormat).unix();
    return stamp;
};

exports.ConvertIdByTimeStamp = (timeStamp) => {
    var id = (timeStamp).toString(16);
    return id;
};

exports.WeekOfDay = (timeStamp) => {
    return parseInt(moment(timeStamp).format('d'));
};
